src/components/transactions/SingleTx/index.tsx

```diff
@@ -0,0 +1,60 @@
+import { CircularProgress } from '@mui/material'
+import ErrorMessage from '@/components/tx/ErrorMessage'
+import useChainId from '@/hooks/useChainId'
+import { useRouter } from 'next/router'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import useAsync from '@/hooks/useAsync'
+import { DateLabel, getTransactionDetails, Transaction, TransactionDetails } from '@gnosis.pm/safe-react-gateway-sdk'
+import { sameAddress } from '@/utils/addresses'
+import { ReactElement } from 'react'
+import { makeDateLabelFromTx, makeTxFromDetails } from '@/utils/transactions'
+import { TxListGrid } from '@/components/transactions/TxList'
+import TxDateLabel from '@/components/transactions/TxDateLabel'
+import ExpandableTransactionItem from '@/components/transactions/TxListItem/ExpandableTransactionItem'
+
+const SingleTxGrid = ({ txDetails }: { txDetails: TransactionDetails }): ReactElement => {
+  const tx: Transaction = makeTxFromDetails(txDetails)
+  const dateLabel: DateLabel = makeDateLabelFromTx(tx)
+
+  return (
+    <TxListGrid data-testid="single-tx-grid">
+      <TxDateLabel item={dateLabel} />
+      <ExpandableTransactionItem item={tx} txDetails={txDetails} />
+    </TxListGrid>
+  )
+}
+
+const SingleTx = () => {
+  const chainId = useChainId()
+  const router = useRouter()
+  const { id } = router.query
+  const transactionId = Array.isArray(id) ? id[0] : id
+  const { safe, safeAddress } = useSafeInfo()
+
+  const [txDetails, txDetailsError, loading] = useAsync<TransactionDetails>(
+    () => {
+      if (!transactionId) return
+      return getTransactionDetails(chainId, transactionId)
+    },
+    [transactionId, safe.txQueuedTag, safe.txHistoryTag],
+    false,
+  )
+  const isCurrentSafeTx = sameAddress(txDetails?.safeAddress, safeAddress)
+
+  const error =
+    transactionId && !isCurrentSafeTx
+      ? new Error(`Transaction  with id ${transactionId} not found in this Safe`)
+      : txDetailsError || new Error("Couldn't retrieve the transaction details. Please review the URL.")
```

Comment:
Why are you falling back to `new Error` unconditionally? If there's no error, it should display any message. It's either loading, or there's an error.

Reply:
Reason is when the URL is malformed there is no TransactionDetails request but we could still inform that it is worth rechecking the URL. Should I simply remove this fallback? 

<img width="783" alt="Screenshot 2022-09-23 at 14 58 32" src="https://user-images.githubusercontent.com/32431609/191965935-098f2b3b-49ff-47c7-9cc3-69ce2faae121.png">

Reply:
What is a "malformed" URL? Missing id? If you know the id is missing, you can use it is as a condition. As it stands, you're just saying "show data if there's data, otherwise show this made up error".